File: Ruby/pangram.rb
def input(sentence)
  sentence.replace(gets.chomp)
end

def output(sentence)
  print check(sentence).to_s
end

def check(sentence)
  letters = Hash.new(0)
  sentence.each_char { |s| letters[s] += 1 }
  isPan = true
  ("a".."z").each do |c|
    if letters[c] == 0
      puts "There is no #{c}"
      isPan = false
    end
  end
  isPan
end

sentence = ""
input(sentence)
output(sentence)

File: C++/pangram.cpp
#include <iostream>
#include <string>
#include <map>

void input(std::string* S);
bool check(std::string sentence);
void output(std::string sentence);

int main(){
    std::string sentence;
    input(&sentence);
    output(sentence);
}

void input(std::string* sentence){
    getline(std::cin, *sentence);
}

void output(std::string sentence){
    std::cout << std::boolalpha << check(sentence);
}

bool check(std::string sentence){
    std::map <char, int> letters;
    for (char s : sentence){
        letters[s]++; 
    }
    bool isPan = true;
    for(char c = 97; c <= 122; c++){
        if(letters[c] == 0){
            std::cout << "There is no " << c << std::endl;
            isPan = false;
        }
    }
    return isPan;
}